src/DynamoPackages/Package.cs

```diff
@@ -338,15 +353,32 @@ internal IEnumerable<PackageAssembly> EnumerateAssembliesInBinDirectory()
             // Use the pkg header to determine which assemblies to load and prevent multiple enumeration
             // In earlier packages, this field could be null, which is correctly handled by IsNodeLibrary
             var nodeLibraries = Header.node_libraries;
+            //node view libraries specify entry points which ONLY contain node view customizatons.
+            var nodeViewLibraries = Header.node_view_libraries;
+            nodeViewLibraries = nodeViewLibraries == null ? new string[] { } : nodeViewLibraries;
+
+            if (nodeLibraries != null)
+            {
+                //if an assembly is somhehow listed as both a node_view_library and a node_library - then treat this as a malformed package.
+                foreach (var nvl in nodeViewLibraries)
+                {
+                    if (nodeLibraries.Contains(nvl))
+                    {
+                        throw new LibraryLoadFailedException(BinaryDirectory, $"package header specified {nvl} as both a node_library, and node_view_library");
```

Comment:
Why have this restriction?

Reply:
please see my comment below https://github.com/DynamoDS/Dynamo/pull/11891#issuecomment-889391358